knowledge_agent/github.py

```
@@ -0,0 +1,215 @@
+    @staticmethod
+    def fetch_file_content(token: str, owner: str, repo: str, file_path: str) -> Optional[str]:
+        try:
+            with httpx.Client(timeout=10.0) as client:
+                res = client.get(f"{GITHUB_API_BASE}/repos/{owner}/{repo}/contents/{file_path}", headers=GitHubClient._get_headers(token))
+                res.raise_for_status()
+                data = res.json()
+                if "content" in data and data.get("encoding") == "base64":
+                    decoded_bytes = base64.b64decode(data["content"])
+                    return decoded_bytes.decode("utf-8", errors="replace")
+                return None
+        except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)